From 110364cff7124f882746f35b29e69c8ca9dd0289 Mon Sep 17 00:00:00 2001
From: Tamal <tasaha@habana.ai>
Date: Mon, 18 Mar 2024 12:03:00 +0200
Subject: [PATCH] [SW-161727] gaudi3: update mbist wait time

This patch updates mbist execution wait time in ppboot.
The wait time value comes from MBIST table, prepared by
DFT team.

Change-Id: I0b7e48c39ef416d624d50a506bf927c53acd9d17
Signed-off-by: Tamal <tasaha@habana.ai>
---

diff --git a/arch/gaudi3/arc/mbist.c b/arch/gaudi3/arc/mbist.c
index 602d16f..201a5e2 100644
--- a/arch/gaudi3/arc/mbist.c
+++ b/arch/gaudi3/arc/mbist.c
@@ -17,8 +17,11 @@
 
 static inline void gaudi3_mbist_wait(u32 pclk)
 {
-	/* TODO: Validate in HW, then enable it */
-	/* udelay((pclk >> 6) + (pclk >> 5)); */
+	/* Reference from DFT team: IST_WAIT(10) = 1000 ns  or 1 us delay,
+	 * IST_WAIT(uint32_t pclk_cycles) = usleep(pclk_cycles * PCLK_PERIOD_IN_USEC);
+	 */
+
+	(pclk % 10) ? udelay((pclk / 10) + 1) : udelay(pclk / 10);
 }
 
 int hl_ppboot_sram_self_repair_mismatch_logic(u32 ist_addr, u32 *data,
